src/pip/_internal/utils/compat.py

```diff
@@ -60,27 +57,29 @@
         cache_from_source = None
 
     uses_pycache = cache_from_source is not None
+else:
+    uses_pycache = True
+    from importlib.util import cache_from_source
 
 
-if sys.version_info >= (3, 5):
-    backslashreplace_decode = "backslashreplace"
-else:
+if PY2:
     # In version 3.4 and older, backslashreplace exists
```

Comment:
This comment is a bit confusing in a Python 2 only branch. Maybe reword it to avoid referring to 3.4?

Reply:
Good idea, done in https://github.com/pypa/pip/pull/6782/commits/97f719cee245efe4cd5f4d18360507bdaea93285.